.devcontainer/setup.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+env_variables=$(yq eval '.env | to_entries | .[] | .key + "=" + .value' .github/workflows/end2end.yaml | sed 's/\${{[^}]*}}//g') && export $env_variables
+export GIT_ACCESS_TOKEN=${GITHUB_TOKEN}
+export E2E_IMAGE_TAG=latest
+
+echo $REGISTRY_PASSWORD | docker login registry.scality.com -u $REGISTRY_LOGIN --password-stdin
+    array_length=`yq ".runs.steps | length - 1" .github/actions/deploy/action.yaml`
+    for i in $(seq 0 $array_length); do
+        step=`yq ".runs.steps[$i]" .github/actions/deploy/action.yaml`
+        working_dir=`yq ".runs.steps[$i].working-directory" .github/actions/deploy/action.yaml`
+        run_command=`yq ".runs.steps[$i].run" .github/actions/deploy/action.yaml`
+
+        # We don't want to run `run-e2e-test.sh` because it is used for linting here, user will run it manually if needed after deployment
+        # We can't run `configure-e2e.sh` here because it needs an image that is not yet built and sent to kind, will be run after
+        (
+            if [[ "$run_command" != "null" && "$run_command" != *"configure-e2e.sh"* && "$run_command" != *"run-e2e-test.sh"* ]]
+            then
+                if [ "$working_dir" != "null" ]
+                then
+                    echo "Changing working dir: $working_dir"
+                    cd $working_dir
+                fi
+                echo "Run command: $run_command"
+                while IFS= read -r line; do
+                    eval $line
+                done <<< "$run_command";
+            fi
+        )
+    done
+
+(
+    cd tests/zenko_tests
+
+    envsubst < 'e2e-config.yaml.template' > 'e2e-config.yaml'
+    if [[ "${ENABLE_RING_TESTS}" == "false" ]]; then
+    yq -i 'del(.locations[] | select(.locationType == "location-scality-ring-s3-v1"))' e2e-config.yaml
+    fi
+    docker build -t $E2E_IMAGE_NAME:$E2E_IMAGE_TAG .
```

Comment:
we may want to add `--cache-from` params, to speed up the build:

```suggestion
    docker build --cache-from $E2E_IMAGE_NAME:$E2E_IMAGE_TAG --cache-from cache-from: type=gha,scope=end2end-test -t $E2E_IMAGE_NAME:$E2E_IMAGE_TAG .
```

(try to read cache both from gha, same scope as in workflow, and from the image which may have been built already...)

Reply:
I don't think the codespace token will have access to use the github actions cache.

Reply:
@KillianG could you try this?

Reply:
This unfortuneatly does not work :(